### PR TITLE
WIP: Ica acct auth

### PR DESCRIPTION
**Pull request by yaruwangway**
This PR provides a new module `mauth` and its tutorial to gaia. `mauth` brings interchain account feature to gaia and allows any msg types.
 
- do we rename `mauth` ?

### CODE REVIEW

File: docs/hub-tutorials/modules/mauth/mauth-module-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  },
+  "timeout_timestamp": "1641572037493534000"
+}
+```
+step 2: use interchain account to execute the ibc transfer in the json file:
+```shell
+gaiad tx icamsgauth submit-tx send-raw.json --connection-id connection-1 --counterparty-connection-id connection-1 --from a --home test-1 --gas 150000
+```
+The ibc/stake will changed from the denom of `0ibc/3C3D7B3BE` to `stake` when the token is back to a on chain test-1.
+ You might need to kill hermes and restart to receive the tokens.
```
Comment (yaruwangway) [inline docs/hub-tutorials/modules/mauth/mauth-module-tutorial.md]: I am not sure about `you might need to kill hermes and restart to receive the tokens`  @glnro , when you try, could you please confirm ? Thank you!

Comment (glnro) [reply docs/hub-tutorials/modules/mauth/mauth-module-tutorial.md]: Sure I'll take a look

File: docs/hub-tutorials/modules/mauth/init_test_1.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
...gVote\", \"/cosmos.vesting.v1beta1.BaseVestingAccount\", \"/cosmos.vesting.v1beta1.DelayedVestingAccount\", \"/cosmos.vesting.v1beta1.ContinuousVestingAccount\", \"/cosmos.vesting.v1beta1.PeriodicVestingAccount\", \"/cosmos.vesting.v1beta1.PermanentLockedAccount\", \"/cosmos.auth.v1beta1.BaseAccount\", \"/cosmos.auth.v1beta1.ModuleAccount\", \"/cosmos.vesting.v1beta1.ContinuousVestingAccount\", \"/cosmos.vesting.v1beta1.DelayedVestingAccount\", \"/cosmos.vesting.v1beta1.PeriodicVestingAccount\", \"/cosmos.vesting.v1beta1.PermanentLockedAccount\", \"/ibc.lightclients.solomachine.v2.Header\", \"/ibc.lightclients.tendermint.v1.Header\", \"/ibc.core.connection.v1.Counterparty\", \"/ibc.core.channel.v1.Channel\", \"/cosmos.tx.v1beta1.Tx\", \"/cosmos.evidence.v1beta1.Equivocation\", \"/cosmos.bank.v1beta1.Supply\", \"/cosmos.params.v1beta1.ParameterChangeProposal\", \"/ibc.core.client.v1.ClientUpdateProposal\", \"/ibc.core.client.v1.UpgradeProposal\", \"/cosmos.upgrade.v1beta1.SoftwareUpgradeProposal\", \"/cosmos.upgrade.v1beta1.CancelSoftwareUpgradeProposal\", \"/cosmos.distribution.v1beta1.CommunityPoolSpendProposal\", \"/cosmos.gov.v1beta1.TextProposal\"]%g' $HOME1/config/genesis.json
```
Comment (yaruwangway) [inline docs/hub-tutorials/modules/mauth/init_test_1.sh:58]: make the same format for `sed`

File: docs/hub-tutorials/modules/mauth/mauth-module-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
+# Mauth module
+
+## Backgroud introduction
+Mauth module allows all the message types from cosmos-sdk.
+
+[insert introduction here]
+
+In the following tutorial, we will practice how to use interchain account through [mauth module](https://github.com/cosmos/gaia/tree/ica-acct-auth/x/mauth) of gaia.
+## Setup preparation
+We will run two cosmos-sdk chains (control chain:test-1 and host chain: test-2), and a relayer to connect these two chains. We will create an account on chain test-1 and call it `a`, and register an interchain account(ica) for `a` on chain test-2, we will create a normal account `b` on chain tst-2 as well. Through these 3 account, we can test:
```
Comment (okwme) [inline docs/hub-tutorials/modules/mauth/mauth-module-tutorial.md]: ``
```suggestion
We will run two cosmos-sdk chains (control chain: `test-1` and host chain: `test-2`), and a relayer to connect these two chains. We will create an account on chain `test-1` and call it `a`, and register an interchain account(ica) for `a` on chain `test-2`, we will create a normal account `b` on chain `test-2` as well. Through these 3 account, we can test:
```

File: docs/hub-tutorials/modules/mauth/mauth-module-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+ - `a` can control `a`'s `ica` to transfer tokens to an account on chain test-2.
```
Comment (okwme) [inline docs/hub-tutorials/modules/mauth/mauth-module-tutorial.md]:
```suggestion
 - `a` can control `a`'s `ica` to transfer tokens to an account on chain `test-2`.
```

File: docs/hub-tutorials/modules/mauth/mauth-module-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+cp  ./target/release/hermes $HOME/.cargo/bin
+```
+#### Create interchain connection and start hermes
+```shell
+source hermes/rly-variables.sh
+sh hermes/rly-restore-keys.sh
+sh hermes/rly-create-conn.sh
+sh hermes/rly-start.sh
+```
+Now, you are running two cosmos-sdk chain at background, and a hermes relayer. Open a new terminal and config gaia home for the two chains. Then you are ready to solve the [interchain account exercise](##Interchain account exercise).
```
Comment (okwme) [inline docs/hub-tutorials/modules/mauth/mauth-module-tutorial.md]:
```suggestion
Now, you are running two cosmos-sdk chain in the background, and a hermes relayer. Open a new terminal and config gaia home for the two chains. Then you are ready to solve the [interchain account exercise](##Interchain account exercise).
```

File: go.mod [hunk 6]
```diff
@@ -3,17 +3,23 @@ module github.com/cosmos/gaia/v6
 go 1.17
 
 require (
-	github.com/cosmos/cosmos-sdk v0.44.3
-	github.com/cosmos/ibc-go/v2 v2.0.0
```
Comment (yaruwangway) [inline go.mod:7]: @okwme

**Comment by nooomski** [comment]
I've cleaned up the tutorial and added an intro to ICA.
Still wondering if we need to rename both the module and the command for this. Using `mAuth`, it's not really clear this is for ICA. Why not call it `icaClient`?

Additionally, `icamsgauth` as a command is too long in form and too narrow in scope. Why not just use `ica` ?

Any thoughts @okwme @glnro @yaruwangway ?

**Comment by nooomski** [comment]
Oh, also wanted to ask @yaruwangway if perhaps we should add some basic setup instructions to clone the repo and whatnot?

**Pull request closed without merging**